### PR TITLE
Update buildtools to allow generate resources code in intermediate folder.

### PR DESCRIPTION
**Pull request by tarekgh**
We are going to remove all checked in SR.cs in the corefx tree and we'll always generate such files on demand. this change will make handling the resources easier as no need to add any source files to the projects

### CODE REVIEW

**Comment by tarekgh** [comment]
@weshaggard  @davkean could you have a quick look?

File: src/Microsoft.DotNet.Build.Tasks/GenerateResourcesCode.cs [hunk 1]
```diff
@@ -58,25 +53,13 @@ public override bool Execute()
...
-                string message = "Resources code generation is failed\n" + e.Message;
-
-                if (e is System.UnauthorizedAccessException)
-                {
-                    message = message + string.Format("\nThe generated {0} file needs to be updated but the file is read-only.", OutputSourceFilePath);
-                }
-
-                Log.LogError(message);
-
+                Log.LogError("Resources code generation is failed\n" + e.Message);
```
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/GenerateResourcesCode.cs]: "Failed to generate the resource code with error:\n"

**Comment by weshaggard** [comment]
Other than the error message change it looks good to me.

File: src/Microsoft.DotNet.Build.Tasks/Targets/resources.targets [hunk 2]
```diff
@@ -23,16 +21,35 @@
...
         ResxFilePath="$(StringResourcesPath)" 
-        IntermediateFilePath="$(IntermediateResOutputFileFullPath)" 
-        OutputSourceFilePath="$(ResourceSRPath)" 
+        OutputSourceFilePath="$(IntermediateResOutputFileFullPath)" 
         AssemblyName="$(AssemblyName)" />
 
     <ItemGroup>
+      <!-- The following Compile element has to be included dynamically inside the Target otherwise intellisense will not work -->
+      <Compile Include="$(IntermediateResOutputFileFullPath)">
+          <Visible>true</Visible>
```
Comment (davkean) [inline src/Microsoft.DotNet.Build.Tasks/Targets/resources.targets]: <Visible> is not going to work. The visual tree is calculated statically at project load, you cannot control the visibility after that.

File: src/Microsoft.DotNet.Build.Tasks/GenerateResourcesCode.cs (hunk 1 above)
Comment (tarekgh) [inline src/Microsoft.DotNet.Build.Tasks/GenerateResourcesCode.cs]: I'll fix the message.

File: src/Microsoft.DotNet.Build.Tasks/Targets/resources.targets (hunk 2 above)
Comment (tarekgh) [inline src/Microsoft.DotNet.Build.Tasks/Targets/resources.targets]: I think you are right. I re-checked that and is not visible. I'll remove <Visible> element

**Comment by tarekgh** [comment]
Thanks Wes and David. I'll merge it.

**Pull request merged**